packages/common/src/models/account.ts

```diff
@@ -65,27 +65,17 @@ export function approve(
   const transferAuthority = existingTransferAuthority || new Account();
   const delegateKey = delegate ?? transferAuthority.publicKey;
 
-  const instruction = Token.createApproveInstruction(
-    tokenProgram,
-    account,
-    delegateKey,
-    owner,
-    [],
-    amount,
+  instructions.push(
+    Token.createApproveInstruction(
+      tokenProgram,
+      account,
+      delegate ?? transferAuthority.publicKey,
+      owner,
+      [],
+      amount,
+    ),
   );
 
-  // Temp. workaround for a bug in Token.createApproveInstruction which doesn't add the delegate account to signers
```

Comment:
I dont understand...has this bug been fixed?

Reply:
No, this workaround was wrong and only coincidently fix our problem. In the approve transaction the delegate account is a beneficiary and hence doesn’t have to sign the transaction. The real issue was the fact that the transferAuthority in our transaction was not flagged as signer. 
This workaround worked because web3 api collects all singing accounts from all transaction and incidentally added transferAuthority to the list.

Reply:
roger